### DISCUSSION TITLE
Auction using Seaport

### DISCUSSION THREAD

**Discussion by Yara1990** [ProjectOpenSea/seaport#1242] [General]
I am in the process of developing an NFT marketplace using Seaport and a custom conduit. I am currently working on implementing the auction process described below:

1. Sellers can list an NFT on Seaport and set a reserve price for it.
2. Once a bid is received, a 24-hour live auction begins for NFTs with a Reserve Auction price.
3. Buyers can make offers on the NFT.
4. The highest bit is accepted automatically when the 24 hr window is expired.

However, I'm unsure about the architecture required for this feature. I would greatly appreciate any assistance in planning the development of this functionality. Specifically, I need guidance on which Seaport functions to consider and whether it's necessary to store bids offchain or if I can utilize the conduit/zone for this purpose.

**Comment by ryanio** [comment]
The way we do this on OpenSea is we set the order's zone to an EOA that is allowed to execute the order. Then we collect bids off chain, and when the auction is due to be closed we'll send a tx from the EOA to match the auction with the highest bid.

You could also develop an on chain zone to handle bids automatically accepting the highest one when the time is right but then bids would have to be validated or provided on chain, and someone still has to send a transaction to settle the auction.

**Reply by nonameb3** [reply]
How do I create order that can receive dynamic amount?
